odoo/netsvc.py

```diff
@@ -39,7 +36,8 @@ def emit(self, record):
         if not dbname:
             return
         with tools.ignore(Exception), tools.mute_logger('odoo.sql_db'), sql_db.db_connect(dbname, allow_uri=True).cursor() as cr:
-            cr.autocommit(True)
+            # preclude risks of deadlocks
```

Comment:
any particular reason for dropping the autocommit mode mysteriously added by d8d3b38cbdc2f73d3a4e41df9781ceb2defd37f8 ?

Reply:
Since SET LOCAL is scoped to the current transaction, in autocommit it's not going to do anything (as at best the transaction ends immediately). The alternative is `SET` but then it needs to be explicitely re-set or the connection remains "configured" until it's closed.

Reply:
But of course! I don't think there was any hard requirements for using `autocommit` (and @KangOl does not remember either), especially as we only have a single statement to execute. So it should be safe to abandon it, and easier than restoring the `statement_timeout` afterwards indeed. :+1: